test/Constraints/closures.swift

```diff
@@ -202,7 +202,7 @@ func acceptNothingToInt (_: () -> Int) {}
 func testAcceptNothingToInt(ac1: @autoclosure () -> Int) {
   // expected-note@-1{{parameter 'ac1' is implicitly non-escaping because it was declared @autoclosure}}
   acceptNothingToInt({ac1($0)})
-  // expected-error@-1{{cannot convert value of type '(_) -> Int' to expected argument type '() -> Int'}}
+  // expected-error@-1{{contextual closure type '() -> Int' expects 0 arguments, but 1 were used in closure body}}
```

Comment:
For the case where contextual type having empty parameter list (ex: `() -> Int`), I added a constraint where the number of elements has to be greater than or equal to 2. (at TypeCheckConstraints.cpp:1537). The rational behind is because in the case of `() -> Int`, the parameter would be derived as a TupleType with 0 elements but not ParenType, and for the case of `(Int) -> Int` it would be a sugared ParenType. And for the case empty tuple it would fall through and catched by TypeCheckPattern.cpp:1573,  I cannot handle the case of empty tuple and early return is because it is still likely the closure has zero input, and that would be best delegated to `coerceParameterListToType`